test-frame-metrics-collector/src/main/java/io/skodjob/testframe/MetricsCollector.java

```diff
@@ -0,0 +1,565 @@
+            return type;
+        }
+
+        public void setType(Type type) {
+            this.type = type;
+        }
+    }
+
+    /**
+     * Collects metrics from all relevant pods within the configured namespace using the default timeout.
+     * This method is a convenience wrapper around {@link #collectMetricsFromPods(long)} with the timeout
+     * set to {@link TestFrameConstants#GLOBAL_TIMEOUT_MEDIUM}.
+     *
+     * <p>This simplifies calls to the metrics collection process when the default timeout is appropriate.
+     * The method orchestrates the metric collection process, handling retries, exceptions, and aggregates
+     * metrics data into a map. It uses the pre-configured timeout defined in {@code TestFrameConstants}.
+     *
+     * @throws MetricsCollectionException if the collection fails due to no data or errors during execution.
+     *                                    This includes scenarios where no pods are found, data is incomplete,
+     *                                    or other errors occur as defined by the implementation of the metric
+     *                                    collection logic.
+     */
+    public final void collectMetricsFromPods() throws MetricsCollectionException {
+        collectMetricsFromPods(Duration.ofSeconds(TestFrameConstants.GLOBAL_TIMEOUT_MEDIUM).toMillis());
+    }
+
+    /**
+     * Collects metrics from each pod that matches the component's label selector immediately without waiting
+     * for data availability checks. This method is useful for scenarios where immediate data retrieval is necessary
+     * and the readiness of data is either ensured or non-critical.
+     *
+     * @return                              A map containing the collected metrics, associated by pod name.
+     * @throws MetricsCollectionException   if errors occur during the collection process.
+     */
+    public final Map<String, String> collectMetricsFromPodsWithoutWait() {
+        final Map<String, String> map = new HashMap<>();
+        final Map<String, String> errorMap = new HashMap<>(); // Store errors separately
+
+        final List<Pod> pods = getKubeClient()
+            .pods()
+            .inNamespace(namespaceName)
+            .withLabelSelector(component.getLabelSelector())
+            .list()
+            .getItems();
+        if (pods.isEmpty()) {
+            LOGGER.info("No pods found with the specified label selector.");
+            return map;
+        }
+
+        for (final Pod p : pods) {
+            final String podName = p.getMetadata().getName();
+            String podIP = p.getStatus().getPodIP();
+            if (TestFrameEnv.IP_FAMILY.equals(TestFrameEnv.IP_FAMILY_VERSION_6)) {
+                podIP = "[" + podIP + "]";
+            }
+
+            try {
+                final String metrics = collectMetrics(podIP, podName);
+                map.put(podName, metrics);
+                LOGGER.info("Collected metrics from {}: {}", podName, metrics);
```

Comment:
```suggestion
                LOGGER.info("Finished metrics collection from {}: {}", podName, metrics);
```

What `metrics` will contains in that case? From code it seems that everything collected from the pod will be in that output. If so, I would put it into debug level for sure and on info just keep info about finishing the collecting.

Reply:
Cool thanks will do.